components/eamxx/src/share/field/field_layout.hpp

```diff
@@ -59,6 +59,10 @@ inline std::string e2str (const LayoutType lt) {
  */
 
 class FieldLayout {
+
+  void add_dim (const FieldTag t, const int extent, const std::string& name,
```

Comment:
Would you mind moving the decl of this method down in the protected section?

When possible, I feel like internal stuff of a class should be down below, so that ppl opening the header in search of API info don't waste scroll space on the internals...

Reply:
No problem! Did you want it protected instead of private?

Reply:
I usually go with protected unless there's a good reason to make it private. Most likely no other class will inherit from this (there are no virtual methods), so either one works. But there's already a protected section.

Reply:
Should be addressed in https://github.com/E3SM-Project/E3SM/pull/7261/commits/30954e2ecabb61f7289c98739ceee60b630b9982